int8 req
---
string results